src/main/java/com/fundation/search/model/Search.java

```diff
@@ -74,15 +111,193 @@
+        Object compareValue = new Object();
+
+        try {
+            compareValue = StorageUnit.class.getMethod(method).invoke(item);
+        }
+        catch (NoSuchMethodException e) {
+            e.printStackTrace();
+        }
+        catch (IllegalAccessException e) {
+            e.printStackTrace();
+        }
+        catch (InvocationTargetException e) {
+            e.printStackTrace();
+        }
+
+        switch (index) {
+            case 0:
+                if (((Date) compareValue).compareTo(criteria.get(index)) == 0) {
+                    return true;
+                }
+                break;
+
+            case 1:
+                if (((Date) compareValue).compareTo(criteria.get(index)) < 0) {
+                    return true;
+                }
+                break;
+
+            case 2:
+                if (((Date) compareValue).compareTo(criteria.get(index)) > 0) {
+                    return true;
+                }
+                break;
+
+            case 3:
+                if (((Date) compareValue).compareTo(criteria.get(index)) <= 0) {
+                    return true;
+                }
+                break;
+
+            case 4:
+                if (((Date) compareValue).compareTo(criteria.get(index)) >= 0) {
+                    return true;
+                }
+                break;
+        }
+
+        return false;
+    }
+
+    public Boolean compareLongs(Map<Integer, Long> criteria, StorageUnit item, String method) {
+
+        Integer index = (int) criteria.keySet().toArray()[0];
+        Object compareValue = new Object();
+
+        try {
+            compareValue = StorageUnit.class.getMethod(method).invoke(item);
+        }
+        catch (NoSuchMethodException e) {
+            e.printStackTrace();
+        }
+        catch (IllegalAccessException e) {
+            e.printStackTrace();
+        }
+        catch (InvocationTargetException e) {
+            e.printStackTrace();
+        }
+
+        if (compareValue == null) {
+            compareValue = new Long(0);
+        }
+
+        switch (index) {
+            case 0:
+                if ((long) compareValue == criteria.get(index)) {
+                    return true;
+                }
+                break;
+
+            case 1:
+                if ((long) compareValue < criteria.get(index)) {
+                    return true;
+                }
+                break;
+
+            case 2:
+                if ((long) compareValue > criteria.get(index)) {
+                    return true;
+                }
+                break;
+
+            case 3:
+                if ((long) compareValue <= criteria.get(index)) {
+                    return true;
+                }
+                break;
+
+            case 4:
+                if ((long) compareValue >= criteria.get(index)) {
+                    return true;
+                }
+                break;
+        }
+
+        return false;
+    }
+
+
```

Comment:
Would be good to document all the classes and all methods to know what does it do without reviewing all the code.